gossip/server.go

```diff
@@ -147,8 +152,8 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 			if log.V(1) {
 				log.Infof("node %d received %s from node %d (%d fresh)", s.is.NodeID, extractKeys(args.Delta), args.NodeID, freshCount)
 			}
-			if s.closed {
-				return nil, util.Errorf("gossip server shutdown")
+			if !s.stopper.RunTask(func() {}) {
```

Comment:
In this case, a best effort check is sufficient, and this code all goes away when gRPC is brought in.